### DISCUSSION TITLE
[BUG] error running example code on v6d connection

### DISCUSSION THREAD

**Discussion by YifanYuan3** [alibaba/GraphScope#2203] [Q&A] [answered]
**Describe the bug**
Hi, I was trying to play with GraphScope (using the official docker images) from its example code. and when I was trying to load the graph with `graph = load_ogbn_mag()`, the bug will happen saying "IOError" (see the attached screenshot for detail). 

It's a bit similar to #1903 but not the same I guess.
It will be great if you could provide insights and helps on it. Thanks in advance! 

**To Reproduce**
1. enter python3 cli.
2. type `import graphscope`
3. type `from graphscope.dataset import  load_ogbn_mag`
4. type `graph = load_ogbn_mag()
`
**Expected behavior**
function successfully completed with graph loaded.

**Screenshots**
![image](https://user-images.githubusercontent.com/21162870/199880886-f41cb77c-2c38-47fb-a476-44263677e445.png)

**Environment (please complete the following information):**
 - GraphScope version: 0.14.0 -- 0.18.0, within the official docker images
 - OS: Linux Ubuntu 20.4

I also tried v6d in the docker image as well with 'python3 -m vineyard --socket=/tmp/vineyard.sock', and seems it has some problems connecting...
![image](https://user-images.githubusercontent.com/21162870/199881378-7a1e5155-a43e-4f9a-88a7-d34d0c2b2f9c.png)

**Comment by welcome** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! And a maintainer will get back to you shortly!
Please feel free to contact us on [DingTalk](https://h5.dingtalk.com/circle/healthCheckin.html?dtaction=os&corpId=dingb4b76e0fa9ccd8d516fd0f0acdf19f55&68e71190=0fe34ec0&cbdbhh=qwertyuiop), WeChat account(graphscope) or [Slack](http://slack.graphscope.io/). We are happy to answer your questions responsively.

**Comment by sighingnow** [comment, marked as answer]
Looks that you have another application `squid` that is running at the 2379 or 2380 port so etcd cannot be launched.

To resolve the problem, you could either

- kill the squid application the retry
- or, specify a different etcd port when starting graphscope, see `etcd_listening_client_port` and `etcd_listening_peer_port` in our documentation: https://graphscope.io/docs/reference/session.html#session-object

**Reply by YifanYuan3** [reply]
Thanks for your quick response! In fact I checked that neither the host nor the container has `squid` running, and port 2379 and 2380 are free on both host and container as well. But I'm in an enterprise network indeed, which requires proxy setting to access internet. maybe that's the problem? 

Meanwhile I will try explicit setting port number as well and let you know.

**Reply by sighingnow** [reply]
> But I'm in an enterprise network indeed, which requires proxy setting to access internet. maybe that's the problem?

I was assuming accessing localhost network wouldn't be affected by proxy.

> Meanwhile I will try explicit setting port number as well and let you know.

If the issue is caused by the global proxy settings, changing port won't work as well, I guess.

**Reply by YifanYuan3** [reply]
I had such guess because I tried the exactly same thing another university server, which has no specific network proxy setting, and everything works well. let me try to disable proxy setting a bit (but it means I cannot download dataset. but anyway I can manually construct a simple dataset to make sure v6d works properly)

**Reply by sighingnow** [reply]
The dataset can be manually downloaded: https://github.com/alibaba/GraphScope/blob/main/python/graphscope/dataset/ogbn_mag.py#L26

**Reply by YifanYuan3** [reply]
it turns out that this problem is indeed related to proxy setting (but in a weird way). After disabling proxy settings and rebooting the machine, I'm able to connect to the v6d and build a graph in the host's python environment with host graphscope. But the container one still doesn't work (surprisingly!). Thanks so much for your responsive comments, and I will dig deeper with company IT department. Will keep you posted if there is anything I find to make graphscope initialization more robust.:)

**Reply by YifanYuan3** [reply]
BTW, let me attach the proxy settings in `.bashrc` (company name removed): 
```
export http_proxy=http://proxy-server1.xxx.com:911
export https_proxy=http://proxy-server1.xxx.com:912
export ftp_proxy=http://proxy-server1.xxx.com:911
export socks_proxy=http://proxy-server2.xxx.com:1080
export no_proxy=xxx.com,.xxx.com,localhost,127.0.0.1,10.0.0.0/8,192.168.0.0/16,172.16.0.0/12
```

**Reply by YifanYuan3** [reply]
Hi @sighingnow , seems that it indeed has something to do with proxy settings. For example, when I explicitly enforce "0.0.0.0" as "NO_PROXY" in the container, `python3 -m vineyard --socket=/tmp/vineyard.sock` works properly. However, graphscope still doesn't work as I stated in the very beginning. Could you please give some hints on (or point me to) the IP addresses used during graphscope init stage (e.g., k8s communication, etcd, vineyard,.....), so that I can try to explicitly enforce the proxy setting in the container/k8s? Thanks so much!

**Reply by sighingnow** [reply]
Maybe you need to add the local IP address `hostname -i` as `NO_PROXY` as well? I think.

**Reply by YifanYuan3** [reply]
yes! adding "hostname -i" works. Thanks so much! Is there a way I can contribute this experience to the community?

**Reply by sighingnow** [reply]
It would be great if you could submit a pull request to adding the experience to our "FAQ" section!

- English version: https://github.com/alibaba/GraphScope/blob/main/docs/frequently_asked_questions.rst
- Chinese version: https://github.com/alibaba/GraphScope/blob/main/docs/zh/frequently_asked_questions.rst
